azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -107,6 +111,7 @@ def with_aad_application_key_authentication(cls, connection_string, aad_app_id,
         _assert_value_is_valid(app_key)
         _assert_value_is_valid(authority_id)
         kcsb = cls(connection_string)
+        kcsb[kcsb.ValidKeywords.aad_federated_security] = True

```

Comment:
why do you assume it's federated? 
what I could find here https://docs.microsoft.com/en-us/azure/active-directory/hybrid/how-to-connect-fed-whatis explains that federation means it's on-premise active directory vs. the azure active directory...

Reply:
Not quite. It is a common use to use "AAD Federated" as a denotion that we authenticate to one of the AAD tenants.

Reply:
https://docs.microsoft.com/en-us/azure/kusto/api/connection-strings/kusto#application-authentication-properties